### PR TITLE
[617] Production checklist

### PR DESCRIPTION
**Pull request by saliceti**
## Context
Document best practices for production environments

## Guidance to review
Read the new documentation, check content and formatting (use view file to see the rendered mardown)

## Checklist

- [x] I have performed a self-review of my code, including formatting and typos
- [x] I have [cleaned the commit history](https://www.annashipman.co.uk/jfdi/good-pull-requests.html)
- [x] I have added the `Devops` label
- [x] I have attached the pull request to the trello card

### CODE REVIEW

**Comment by saliceti** [comment]
[Add production checklist to service onboarding](https://trello.com/c/M8dh9fDS/617-add-production-checklist-to-service-onboarding)

File: documentation/production-checklist.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
+# Production checklist
+
+For the service to be ready for end users, it must be reliable, performant and sustainable.
+
+## StatusCake
+This is the most essential monitoring as if it alerts, it means users cannot access the site. It monitors both [uptime](https://www.statuscake.com/features/uptime/) and [SSL certificate](https://www.statuscake.com/features/ssl/). Use the [terraform module](https://github.com/DFE-Digital/terraform-modules/blob/main/monitoring/statuscake/README.md) to configure it.
+
+## Multiple replicas
+By default the template deploys only 1 replica for each [kubernetes deployment](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/). This is not sufficient for production as if the container is unavailable, there is no other replica to serve the requests. It may be unavailable because of high usage or simply because the cluster is moving the container to another node. This will happen when the cluster version is updated.
```
Comment (RMcVelia) [inline documentation/production-checklist.md:9]: Do we add a comment on replica memory sizing? Default for prod is 1G but app memory usage should be checked before and after launch.

File: documentation/production-checklist.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+The same should be applied to database clusters. For postgres, set `azure_enable_high_availability` to true. For redis, use a Premium plan.
+
+Note the cost is doubled for postgres, and [much higher](https://azure.microsoft.com/en-gb/pricing/details/cache/) for redis, so this should be used carefully.
+
+## Performance testing
+Simulate load from user traffic to determine the right number of instances and the database plan. This should cover the most typical user journeys. We recommend [K6](https://k6.io/) as it can be deployed to the cluster to minimise latency. Check the example in [teacher pay calculator](https://github.com/DFE-Digital/teacher-pay-calculator/tree/main/load_testing).
+
+If time is short or user traffic is expected to be low, make sure to monitor the application and database usage after launch, and everytime there is a new significant feature. And be ready to scale up.
+
+## Postgres and redis monitoring
```
Comment (RMcVelia) [inline documentation/production-checklist.md:32]: Add postgres backup should be in place for production before launch?

File: documentation/production-checklist.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+Set `azure_enable_monitoring` to true to enable logging, monitoring and alerting. It will alert the infrastructure team by email by default.
+
+## Front door monitoring
+Set `azure_enable_monitoring` to true to enable logging on front door. It is very verbose and costly and should not be used by default. But it can be extremely useful for troubleshooting.
```
Comment (RMcVelia) [inline documentation/production-checklist.md:36]: should it mention this is in the domains/infrastructure module, not the aks module, so is different to postgres and redis monitoring? or should this be in a troubleshooting doc instead?

File: documentation/production-checklist.md (hunk 2 above)
Comment (saliceti) [reply documentation/production-checklist.md:37]: Added

**Review by RMcVelia — APPROVED** [review]

**Pull request merged**